Header UI Updates

mwodahl opened the pull request:
# PR Details

This PR updates the CV-Manager header to reflect the following Figma prototype: https://www.figma.com/proto/QG64ugNXpXMUk2vQ1wgPdZ/CV-Manager-Prototype-January-2025?page-id=2143%3A6061&node-id=2146-7246&viewport=-595%2C-659%2C0.22&t=A1CuORABoO0Qexzk-1&scaling=scale-down&content-scaling=fixed&starting-point-node-id=2146%3A6646

## Description

This PR includes the following updates:
- Tabs now share the same background color as the header
- Organization selection and logout button have been moved to a dropdown menu
- Selected tabs are identified with bold font and a bottom border
- Testing snapshots have been updated

## How Has This Been Tested?

This has been tested locally using docker compose. All existing functionality for the user dropdown menu was verified to work with these changes. Additionally, testing was verified to work using npm test.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ X ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] My changes require new environment variables:
  - [ ] I have updated the docker-compose, K8s YAML, and all dependent deployment configuration files.
- [ ] My changes require updates to the documentation:
  - [ ] I have updated the documentation accordingly.
- [ X ] My changes require updates and/or additions to the unit tests:
  - [ X ] I have modified/added tests to cover my changes.
- [ X ] All existing tests pass.

mwodahl committed:
Change background color, margin, fontWeight on HorizontalTabs.tsx

mwodahl committed:
Update login padding

mwodahl committed:
Update header to include user menu

mwodahl committed:
Header UI testing snapshot updates

mwodahl committed:
Merge branch 'develop' into Update/webapp-header-ui

rgelmis commented on webapp/src/styles/components/HorizontalTabs.tsx:
```diff
@@ -16,18 +14,15 @@ interface TabItemStyledProps {
     fontFamily: 'Arial, Helvetica, sans-serif',
-    fontWeight: 550,
+    fontWeight: 750,
     backgroundColor: theme.palette.primary.main,
```
This should be removed, active links do not need a background color in the new UI

mwodahl replied:
Removed

rgelmis commented on webapp/src/styles/components/HorizontalTabs.tsx:
```diff
@@ -16,18 +14,15 @@ interface TabItemStyledProps {
-  marginBottom: '-1px',
   padding: '0.5rem 0.75rem',
   cursor: 'pointer',
   color: theme.palette.text.primary,
```
Color for inactive tabs should be `#c7c7c7` to differentiate inactive tabs from the styling for an active tab. I recommend updating `index.ts` to use that value as the secondary text color, since it will be used in other places

rgelmis replied:
I'm adding this as a suggested update for #172

mwodahl replied:
Updated to use the secondary text color which has been changed in #172 to `#c7c7c7`

rgelmis commented on webapp/src/styles/components/HorizontalTabs.tsx:
```diff
@@ -16,18 +14,15 @@ interface TabItemStyledProps {
   ...(isActive && {
     fontFamily: 'Arial, Helvetica, sans-serif',
-    fontWeight: 550,
+    fontWeight: 750,
```
Font weight should be 500 for all tabs

mwodahl replied:
Updated

rgelmis commented on webapp/src/components/Header.tsx:
```diff
@@ -65,44 +79,90 @@ const Header = () => {
-        <Paper id="header">
+        <Paper id="header" elevation={0}>
           <Grid2 container alignItems="center" style={{ height: 'fit-content' }}>
             <img id="logo" src={iconPath} alt="Logo" height="90px" />
```
```suggestion
            <img id="logo" src={iconPath} alt="Logo" height="34px" />
```

mwodahl replied:
Updated

rgelmis commented on webapp/src/components/Header.tsx:
```diff
@@ -1,4 +1,4 @@
+import React, { useEffect, useMemo, useState } from 'react'
 import Grid2 from '@mui/material/Grid2'
 import { useSelector, useDispatch } from 'react-redux'
 import EnvironmentVars from '../EnvironmentVars'
```
```suggestion
import EnvironmentVars from '../EnvironmentVars'
import Box from '@mui/material/Box'
```

Importing this so it can be used to further refine layout

rgelmis commented on webapp/src/components/Header.tsx:
```diff
@@ -65,44 +79,90 @@ const Header = () => {
+                    {organizationName}
+                  </Typography>
+                </Typography>
+              </Button>
```
```suggestion
         <Grid2 container alignItems="center">
            <Grid2>
              <img id="logo" src={iconPath} alt="Logo" height="34px" />
            </Grid2>
            <Grid2 size="grow">
              <h2 id="header-text">{EnvironmentVars.DOT_NAME} CV Manager</h2>
            </Grid2>
            <Grid2>
              <Button
                id="userInfoButton"
                aria-controls={open ? 'user-menu' : undefined}
                aria-haspopup="true"
                aria-expanded={open ? 'true' : undefined}
                variant="text"
                color="primary"
                endIcon={<ArrowDropDown color="info" sx={{ marginLeft: 1}} />}
                onClick={handleMenuOpen}
                className="user-info-btn"
              >
                <Box display="flex" flexDirection="column" alignItems="start">
                <Typography fontSize="small" color={theme.palette.text.primary} className="capital-case">
                  {userName}
                  <Typography fontSize="small" color={theme.palette.text.primary} className="capital-case">
                    {organizationName}
                  </Typography>
                </Typography>
                </Box>
              </Button>
```

There's a lot going on here- I had to refine the layout so we are have less hardcoded values and so all the header elements are within the same parent. The padding updates weren't applying to the user info button, so that needed to be updated too. I increased the margin for the icon and changed the font size for the user info button so the header doesn't take up so much vertical space and matches the prototype better. If you have any questions let me know!

mwodahl replied:
Updated to use this layout

rgelmis commented on webapp/src/components/Header.tsx:
```diff
@@ -65,44 +79,90 @@ const Header = () => {
+                  </MenuItem>
+                </div>
+              </Menu>
             </div>
```
```suggestion
            </Grid2>
```

rgelmis commented on webapp/src/components/Header.tsx:
```diff
@@ -65,44 +79,90 @@ const Header = () => {
+              </Menu>
             </div>
           </Grid2>
         </Paper>
```
I can't comment on line 114 but the `<h1>` element needs to be changed to `<h2>`

mwodahl replied:
The only remaining `<h1>` element is displayed only if a user's login is unsuccessful. Since this page wasn't included in the Figma prototype, I've created a work item to check the page and see what UI updates might be necessary.

rgelmis commented on webapp/src/components/css/Header.css:
```diff
@@ -41,7 +41,7 @@
 }
 
```
```suggestion
.capital-case {
  text-transform: capitalize;
}

.user-info-btn {
  text-align: right;
}
```

mwodahl replied:
Added

rgelmis replied:
`.user-info-btn` still needs to be added, I'm not seeing it in this file

mwodahl replied:
Added `.user-info-btn`

rgelmis commented on webapp/src/components/css/Header.css:
```diff
@@ -41,7 +41,7 @@
 }
 
```
Line 5 needs to be changed to `padding: 8x 24px;`

mwodahl replied:
Updated

rgelmis replied:
One small thing left here- the padding needs to be updated for the horizontal tabs as well. It just needs to match the padding for the header 
![image](https://github.com/user-attachments/assets/15a93ea1-77ac-483d-a644-583cafb78320)

mwodahl replied:
I've updated the padding for horizontal tabs as well as their container to match the prototype

rgelmis commented on webapp/src/components/css/Header.css:
```diff
@@ -41,7 +41,7 @@
 }
 
```
[Line 32](https://github.com/CDOT-CV/jpo-cvmanager/pull/159/files#diff-9f2296e2404990ec06b201991c574b790f87d6589aab8a17ab0e27e70ef0bcf0L32) Remove `margin-left: 10px;`

mwodahl replied:
Removed

rgelmis reviewed (changes requested):
I left some suggestions for you, let me know if you have any questions or if you notice anything I might have missed!

rgelmis commented:
One other small change- On `Dashboard.tsx` I removed line 31 and 33, it appeared to be extraneous given the other changes I suggested in this PR. 
![image](https://github.com/user-attachments/assets/5d2133b7-01b2-4760-9d37-09aad67f882e)

mwodahl committed:
Update tab text color, font weight

mwodahl committed:
Remove unnecessary div element

mwodahl committed:
Update unselected HorizontalTabs font weight

mwodahl committed:
Update header layout to use Grid component

mwodahl committed:
Remove unnecessary fontWeight property

mwodahl committed:
Update header padding

mwodahl committed:
add capital-case class

mwodahl committed:
Remove unnecessary height attribute from unsuccessful login logo

mwodahl commented:
> One other small change- On `Dashboard.tsx` I removed line 31 and 33, it appeared to be extraneous given the other changes I suggested in this PR. ![image](https://private-user-images.githubusercontent.com/111303732/427117488-5d2133b7-01b2-4760-9d37-09aad67f882e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMxOTc1MzksIm5iZiI6MTc0MzE5NzIzOSwicGF0aCI6Ii8xMTEzMDM3MzIvNDI3MTE3NDg4LTVkMjEzM2I3LTAxYjItNDc2MC05ZDM3LTA5YWFkNjdmODgyZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMyOFQyMTI3MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMDRhOGQ3NTkwZmUwZDY2NGNhMWIwZmM4NWY4NDU2ZjFmODZjOGZlMjIxZGQ2NWYzYmVlMmEzZGVlZjNjOGY3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5jEqcKLDlqxlVLzXT0h-Hajv_6WKOos3yZ21p7_ojwg)

Removed this unnecessary `<div>` element

rgelmis reviewed (changes requested):
I noticed 2 small things that need to be resolved, otherwise this looks great!

mwodahl committed:
add user-info-btn to Header.css

mwodahl committed:
Remove logo class from Header.css

mwodahl committed:
Update Horizontal Tab styling

mwodahl committed:
Merge branch 'Update/ui-theme' into Update/webapp-header-ui

mwodahl committed:
Update failing testing snapshots after merging Update/ui-theme

rgelmis commented:
This is looking really good! Just a couple more things:
### User Dropdown
![image](https://github.com/user-attachments/assets/530f8d92-b798-48ef-b531-4e2f6f9f934c)
- Change to use Trebuchet font
- "Organizations" label should be size 12px, secondary text color, all caps
- All text should be secondary text color other than the actively selected organization which should use primary text color
- Logout is an `<li>` element for some reason? Should be an outline button 
![image](https://github.com/user-attachments/assets/7c95ec31-4354-4efc-9ff6-25351ddceea6)

mwodahl committed:
Update divider color to match prototype

mwodahl committed:
Update user dropdown background color, text capitalization, and text color in Header.tsx

mwodahl committed:
Remove unused classes, add capitalize and trebuchet classes

mwodahl committed:
Merge branch 'Update/ui-theme' into Update/webapp-header-ui

mwodahl committed:
add museo-slab font to header text, logout button, user dropdown

mwodahl committed:
Update outdated testing snapshots

rgelmis commented:
lgtm! ✨
